### PR TITLE
Unify ImageInset and Media blocks in FullWidthContent, add 470px size option

### PR DESCRIPTION
**Pull request by Scotchester**
This simplifies our image insertion options in FullWidthText to a single block, now called ContentImage that can work for both inset (floating) and full-width images.

The motivation for doing this was driven by the desire to add a caption field to full-width images (Media), but the block used for that was just a stock Wagtail ImageChooserBlock, so we would've had to create a new StreamBlock to house the image and the caption. It seemed like the simpler path, for both development and user experience, was to combine Media and ImageInset into a single ContentImage block that could support both layout use cases.

Also adds a 470px width option.

This should fulfill the remaining requests in https://[GHE]/CFGOV/platform/issues/2734

## Additions

- A new deprecated-blocks.css file for the Wagtail admin to hold styles dedicated to hiding blocks that are on the path to being completely removed, like Media (also moves the ImageText and LinkBlob hiding styles there)

## Removals

- Media is soft-deprecated by hiding it from the Wagtail admin UI

## Changes

- ImageInset becomes ContentImage, adds options for full width and 470px width

## Testing

1. Pull branch
1. `gulp styles`
1. Add a ContentImage block (just called "Image" in the admin) to a FullWidthContent block
1. Experiment with full width, 470px width, left, right, etc.

## Screenshots

screen shot 2018-05-22 at 16 48 16

## Notes

- To see the diff of the image-inset/content-image template, view just the first commit: 2afcc06

## Todos

- Got some cross-browser testing to do
- Will create migration after approval, to prevent the need to recreate it if other migrations are merged in first 

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [ ] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/molecules/content-image.html [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    value.image_width:    String representing the image width.
+                          Possible values: (170, 270, 470).
+                          Defaults to 270.
+
+    value.image_position: String indicating how to float the image.
+                          Possible values: (left, right).
+                          Defaults to right.
+
+    value.text:           String of caption text to be displayed below image.
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/content-image.html]: Should this description include `(Optional)`?

File: cfgov/jinja2/v1/_includes/molecules/content-image.html [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    value.image:          An ImageBasic atom object containing:
+
+    value.image.upload:   A CFGOVImage object.
+
+    value.is_bottom_rule: Boolean indicating whether the inset has a
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/content-image.html]: should this be prefixed with `has_` not `is_`?

File: cfgov/jinja2/v1/_includes/molecules/content-image.html [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+               m-inset__image
+               {{ 'm-inset__left' if value.image_position == 'left' else '' }}
+               {{ 'm-inset__' ~ ( value.image_width if value.image_width else 270 ) }}
+               {{ 'm-inset__' + image_inset.orientation }}
+               {{ 'm-inset__bottom-rule' if value.is_bottom_rule else '' }}">
+    <div class="m-inset_image-container">
+    {%- if image_alt_value(value.image) %}
+        <img class="m-inset_image"
+             src="{{ image_inset.url }}"
+             srcset="{{ image_inset.url }} {{ image_inset.width }}w,
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/content-image.html]: 👏

File: cfgov/jinja2/v1/_includes/molecules/content-image.html [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+   ========================================================================== #}
+
+{# Image sizes are doubled for retina displays. Images created via Wagtail
+   are never stretched or upscaled and maintain their aspect ratio. #}
+{% set image_inset = image(value.image.upload, 'original') %}
+{% set image_mobile = image(value.image.upload, 'width-1200') %}
+
+<figure class="m-inset
+               m-inset__image
+               {{ 'm-inset__left' if value.image_position == 'left' else '' }}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/content-image.html]: Minor, could include spaces in strings and add `-` within brackets.

File: cfgov/v1/atomic_elements/molecules.py [hunk 5]
```diff
@@ -317,27 +317,32 @@ class Meta:
...
-                                         label='Bottom Rule')
+    image_position = blocks.ChoiceBlock(
+        choices=[('right', 'right'),
+                 ('left', 'left')],
+        default='right',
+        help_text='Does not apply if the image is full-width'
+    )
+    image_width = blocks.ChoiceBlock(
+        choices=[('full', 'full'),
+                 (470, '470px'),
```
Comment (jimmynotjim) [inline cfgov/v1/atomic_elements/molecules.py]: These numbers don't need to be quotes?

File: cfgov/v1/atomic_elements/molecules.py [hunk 6]
```diff
@@ -317,27 +317,32 @@ class Meta:
...
+                 (270, '270px'),
+                 (170, '170px')],
+        default='full',)
+    text = blocks.RichTextBlock(required=False, label='Caption')
+    is_bottom_rule = blocks.BooleanBlock(
+        required=False,
+        default=True,
```
Comment (jimmynotjim) [inline cfgov/v1/atomic_elements/molecules.py]: If it's not required should it have a default?

**Comment by jimmynotjim** [comment]
screen shot 2018-05-22 at 5 06 31 pm

Looks like you need to update the template field to `content-image.html`

File: cfgov/jinja2/v1/_includes/molecules/content-image.html (hunk 1 above)
Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/molecules/content-image.html]: Yeah, I'll add that.

File: cfgov/jinja2/v1/_includes/molecules/content-image.html (hunk 2 above)
Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/molecules/content-image.html]: It _should_, but that's not how it's defined in the Wagtail model, and renaming the field would necessitate a data migration.

File: cfgov/v1/atomic_elements/molecules.py (hunk 5 above)
Comment (Scotchester) [reply cfgov/v1/atomic_elements/molecules.py]: Apparently not!

File: cfgov/v1/atomic_elements/molecules.py (hunk 6 above)
Comment (Scotchester) [reply cfgov/v1/atomic_elements/molecules.py]: If we want it to be on by default, yes. It's a checkbox.

**Comment by Scotchester** [comment]
D'oh. Updated that template value and force-pushed with a rebased branch.

File: cfgov/v1/atomic_elements/molecules.py (hunk 6 above)
Comment (jimmynotjim) [reply cfgov/v1/atomic_elements/molecules.py]: Ohhh, was thinking Boolean was a radio button. Carry on.

**Comment by jimmynotjim** [comment]
That fixed it. Unfortunately decorative images aren't working, there's no padding to make them have a height.

**Comment by jimmynotjim** [comment]
470 setting is displaying at the 270 size 

screen shot 2018-05-23 at 9 30 55 am

**Comment by Scotchester** [comment]
I've rebased and pushed an update to this that fixes the bugs Jimmy originally observed and significantly simplifies things by removing the `background-image` mode of operation. Decorative images will now simply render as `img` elements with empty `alt` attributes.

The only way to handle the `background-image` mode responsively was to include inline media queries within a `style` element, since we had to calculate the height at render time. `style` is not valid within `body`, so I determined that this was a bad approach to take. It was also enormously complicated code, when an empty-alt `img` will do just fine and meet accessibility guidelines, too.

This is ready for review again. Jimmy's on vacation now, so tagging @sebworks and @anselmbradford to review, if possible.

File: cfgov/unprocessed/css/molecules/full-width-media.less [hunk 7]
```diff
@@ -24,10 +24,14 @@
 */
 
 .m-full-width-media {
+    margin-bottom: unit( @grid_gutter-width / @base-font-size-px, em );
```
Comment (sebworks) [inline cfgov/unprocessed/css/molecules/full-width-media.less]: We typically don't add margin to the atomic components.

Comment (Scotchester) [inline cfgov/unprocessed/css/molecules/full-width-media.less]: Ah yes, thanks. I noticed that earlier but forgot to go back and fix it.

Comment (Scotchester) [reply cfgov/unprocessed/css/molecules/full-width-media.less]: Ah yes, thanks. I noticed that earlier but forgot to go back and fix it.

**Comment by sebworks** [comment]
Looks good outside of the margin on the full width text.

File: cfgov/templates/wagtailadmin/css/deprecated-blocks.css [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Hide buttons for the deprecated blocks that InfoUnitGroup replaces.
+ * TODO: Remove when all data is migrated to IUGs and these blocks are totally
```
Comment (anselmbradford) [inline cfgov/templates/wagtailadmin/css/deprecated-blocks.css]: IUGs could be expanded, or on the line above `InfoUnitGroup (IUG)`

File: cfgov/jinja2/v1/_includes/molecules/content-image.html (hunk 1 above)
Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/molecules/content-image.html]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/4177/commits/32f5fcda7d16039558b597d75496a5d12e7f3736

**Pull request merged**